tests/test_dimension.py

```diff
@@ -1288,6 +1288,20 @@ def test_no_index_symbolic(self):
         op = Operator(eq)
         op.cfunction
 
+    @pytest.mark.parametrize('value', [0, 1])
+    def test_constant_as_condition(self, value):
+        x = Dimension('x')
+
+        c = Constant(name="c", dtype=np.int8, value=value)
```

Comment:
Shouldn't this be a Bool if it's a condition by itself?

Reply:
Actually you can enhance the test with a bool as well? Behaviour should be the same?

Reply:
But C doesn't have a core bool type? You have to import `stdbool.h`

Reply:
Convention is to use an int iirc

Reply:
yes, int is OK

Reply:
int8 btw is converted into a char